build.rs

```diff
@@ -1,53 +1,161 @@
 //! Downloads and statically links the `mach-dxcompiler` C library.
 
-use std::env::*;
-use std::path::*;
-use std::process::*;
+use std::env;
+use std::hash::{DefaultHasher, Hash, Hasher};
+use std::process::Command;
+use std::{
+    fs,
+    io::ErrorKind::NotFound,
+    path::{Path, PathBuf},
+};
 
 /// Downloads and links the static DXC binary.
 fn main() {
-    let target = var("TARGET").expect("Failed to get target triple");
-
-    let out_dir = PathBuf::from(var("OUT_DIR").expect("Failed to get output directory"));
-    let file_path = out_dir.join("machdxcompiler.tar.gz");
-    download_url(&get_target_url(&target, static_crt()), &file_path);
+    println!("cargo:rerun-if-changed=build.rs");
```

Comment:
Is this line necessary? I thought that Cargo will always rerun a build script if its source code changes.

Reply:
I think it is redundant, I added this because lots of scripts hava this.